inference-engine/src/gna_plugin/runtime/gna_float_runtime.cpp

```diff
@@ -0,0 +1,70 @@
+using namespace GNAPluginNS::runtime;
+
+
+void FP::infer() {
+    if (!dnn) {
+        THROW_GNA_EXCEPTION << "[GNA FLOAT RUNTIME] not initialized";
+    }
+
+    for (uint32_t i = 0; i < dnn->component.size(); i++) {
+        intel_dnn_component_t *comp = &dnn->component[i];
+        uint32_t *ptr_active_outputs = nullptr;
+        uint32_t num_active_outputs = (comp->orientation_out == kDnnInterleavedOrientation)
+                                      ? comp->num_rows_out : comp->num_columns_out;
+
+        if (i == dnn->component.size() - 1) {  // active list applies to last component
+            ptr_active_outputs = dnn->ptr_active_outputs();
+            num_active_outputs = dnn->num_active_outputs();
+        } else if (i == dnn->component.size() - 2) {  // also applies to last two components when last is PWL
+            if ((dnn->component[i].operation == kDnnAffineOp) && (dnn->component[i + 1].operation == kDnnPiecewiselinearOp)) {
+                ptr_active_outputs = dnn->ptr_active_outputs();
+                num_active_outputs = dnn->num_active_outputs();            }
+        }
+
+        switch (comp->operation) {
+            case kDnnAffineOp :ApplyAffineTransform(comp, ptr_active_outputs, num_active_outputs);
+                break;
+            case kDnnDiagonalOp:ApplyDiagonalTransform(comp);
+                break;
+            case kDnnRecurrentOp:
+                if ((i < dnn->component.size() - 1) && (dnn->component[i + 1].operation == kDnnPiecewiselinearOp)) {
+                    intel_dnn_component_t *comp_pwl = &dnn->component[i + 1];
+                    for (uint32_t j = 0; j < comp->num_rows_in; j++) {
+                        void *ptr_feedbacks =
+                            reinterpret_cast<void *>(reinterpret_cast<int32_t *>(comp->op.recurrent.ptr_feedbacks) + j * comp_pwl->num_columns_out);
+                        ApplyRecurrentTransform(comp, j, ptr_feedbacks);
+                        ApplyPiecewiseLinearTransform(comp_pwl, kDnnFloat, num_active_outputs, j);
+                    }
+                    i++;  // skip next component
+                } else {
+                    THROW_GNA_EXCEPTION << "Missing PiecewiseLinear component after Recurrent component in Propagate!";
+                }
+                break;
+            case kDnnConvolutional1dOp:ApplyConvolutional1DTransform(comp);
+                break;
+            case kDnnPiecewiselinearOp:ApplyPiecewiseLinearTransform(comp, kDnnFloat, num_active_outputs);
+                break;
+            case kDnnMaxPoolOp:ApplyMaxPoolTransform(comp, kDnnFloat);
+                break;
+            case kDnnInterleaveOp:ApplyTranspose(comp);
+                break;
+            case kDnnDeinterleaveOp:ApplyTranspose(comp);
+                break;
+            case kDnnCopyOp:ApplyCopy(comp);
+                break;
+            default:
+                THROW_GNA_EXCEPTION << "[FR RUNTIME] Bad operation " << comp->operation;
+        }
+    }
+}
```

Comment:
where? in exception? disagree

Reply:
file should end with new line. even github gives that "no newline at end of file", so add newline after line 70

Reply:
lets first fix cpp lint - wont fix every new line miss especially no in hpp files. In hpp this make sens in concatenation in cpp - i dont care. Are you aware of any issues with new line in translation unit? Even for hpp this awrning gets removed in c++11

_Just wanted to point any interested parties to James' answer below. While the above answer is still correct for C, the new C++ standard (C++11) has been changed so that this warning should no longer be issued if using C++ and a compiler conforming to C++11._

https://stackoverflow.com/questions/72271/no-newline-at-end-of-file-compiler-warning

Reply:
If it leads to no warnings on any of our build configurations feel free to resolve

Reply:
the only way is to use -Wall to get rid of warnings. Lets plan this activity for new gna-contributors  :1st_place_medal: